### PR TITLE
AM-2732:CA & TS new role mapping for IAC - CTSC

### PR DESCRIPTION
**Pull request by prashanthkumarnb**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/AM-2732

### Change description ###

> NB: dependent on hmcts/am-role-assignment-service#1864

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ *] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolCtscOrgRoleMappingTest.java [hunk 1]
```diff
@@ -71,5 +71,56 @@ void shouldReturnIACCtscMappings(String roleId, String serviceCode, String expec
                 });
     }
 
+    @ParameterizedTest
+    @CsvSource({
+            "10,BFA1,'task-supervisor',Y,N",
+            "10,BFA1,'case-allocator',N,Y",
+            "10,BFA1,'task-supervisor,case-allocator',Y,Y",
+            "9,BFA1,'ctsc-team-leader',N,N"
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolCtscOrgRoleMappingTest.java:79]: Can we expand to cover the other `roleId = 9` scenarios that are being added.

**Review by mattnayler — COMMENTED** [review]
Added some minor comments

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolCtscOrgRoleMappingTest.java (hunk 1 above)
Comment (prashanthkumarnb) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolCtscOrgRoleMappingTest.java:79]: addressed

**Review by mattnayler — DISMISSED** [review]

File: src/main/resources/validationrules/iac/iac-ctsc-mapping.drl [hunk 2]
```diff
@@ -67,3 +65,88 @@ then
...
+      .roleCategory(RoleCategory.CTSC)
+      .roleType(RoleType.ORGANISATION)
+      .roleName("ctsc-team-leader")
+      .grantType(GrantType.STANDARD)
+      .classification(Classification.PUBLIC)
+      .readOnly(false)
+      .attributes(attribute)
+      .authorisations($up.getSkillCodes())
+      .build());
+      logMsg("Rule : iac_ctsc-team-leader_v12");
```
Comment (mikebrownccd) [inline src/main/resources/validationrules/iac/iac-ctsc-mapping.drl:95]: the rule names should use _ not -

Comment (prashanthkumarnb) [reply src/main/resources/validationrules/iac/iac-ctsc-mapping.drl]: addressed

File: src/main/resources/validationrules/iac/iac-ctsc-mapping.drl [hunk 3]
```diff
@@ -67,3 +65,88 @@ then
       .build());
       logMsg("Rule : iac_hmcts_ctsc_caseworker_v11");
 end;
+
+/*
+ * IAC ctsc "ctsc-team-leader" Org role mapping. added for CR:AM2732
+ */
+
+rule "iac_ctsc-team-leader_v12"
```
[the remark this answers is not in the document]
Comment (prashanthkumarnb) [reply src/main/resources/validationrules/iac/iac-ctsc-mapping.drl]: addressed

**Review by mikebrownccd — APPROVED** [review]
LGTM

**Review by mattnayler — APPROVED** [review]

**Pull request merged**